sql-ballerina/Module.md

```diff
@@ -0,0 +1,302 @@
+
+```ballerina
+error? e = dbClient.close();
+```
+Or
+```ballerina
+check dbClient.close();
+```
+
+### Database Operations
+
+Once the client is created, database operations can be executed through that client. This module defines the interface 
+and common properties that are shared among multiple database clients.  It also supports querying, inserting, deleting, 
+updating, and batch updating data.  
+
+#### Creating Tables
+
+This sample creates a table with two columns. One column is of type `int` and the other is of type `varchar`.
+The CREATE statement is executed via the `execute` remote function of the client.
+
+```ballerina
+// Create the ‘Students’ table with the  ‘id’, 'name' and ‘age’ fields.
+sql:ExecutionResult ret = check dbClient->execute("CREATE TABLE student(id INT AUTO_INCREMENT, " +
+                         "age INT, name VARCHAR(255), PRIMARY KEY (id))");
+//A value of type sql:ExecutionResult is returned for 'ret'. 
+```
+
+#### Inserting Data
+
+This sample shows three examples of data insertion by executing an INSERT statement using the `execute` remote function 
+of the client.
+
+In the first example, the query parameter values are passed directly into the query statement of the `execute` 
+remote function.
+
+```ballerina
+sql:ExecutionResult ret = check dbClient->execute("INSERT INTO student(age, name) " +
+                         "values (23, 'john')");
+```
+
+In the second example, the parameter values, which are in local variables are used to parameterize the SQL query in 
+the `execute` remote function. This type of a parameterized SQL query can be used with any primitive Ballerina type 
+like `string`, `int`, `float`, or `boolean` and in that case, the corresponding SQL type of the parameter is derived 
+from the type of the Ballerina variable that is passed in. 
+
+```ballerina
+string name = "Anne";
+int age = 8;
+
+sql:ParameterizedQuery query = `INSERT INTO student(age, name)
+                                values (${age}, ${name})`;
+sql:ExecutionResult ret = check dbClient->execute(query);
+```
+
+In the third example, the parameter values are passed as a `sql:TypedValue` to the `execute` remote function. Use the 
+corresponding subtype of the `sql:TypedValue` such as `sql:Varchar`, `sql:Char`, `sql:Integer`, etc, when you need to 
+provide more details such as the exact SQL type of the parameter.
+
+```ballerina
+sql:VarcharValue name = new ("James");
+sql:IntegerValue age = new (10);
+
+sql:ParameterizedQuery query = `INSERT INTO student(age, name)
+                                values (${age}, ${name})`;
+sql:ExecutionResult ret = check dbClient->execute(query);
+```
+
+#### Inserting Data With Auto-generated Keys
+
+This example demonstrates inserting data while returning the auto-generated keys. It achieves this by using the 
+`execute` remote function to execute the INSERT statement.
```

Comment:
```suggestion
`execute` remote function to execute the `INSERT` statement.
```

Reply:
Change Added in https://github.com/ballerina-platform/module-ballerina-sql/pull/228/commits/2656f12c62276e15a256c133465cb60304d0b041